controllers/host/host_controller.go

```diff
@@ -1729,37 +1729,36 @@ func (r *HostReconciler) ReconcileExistingHost(client *gophercloud.ServiceClient
 	}
 
 	inSync := r.CompareAttributes(profile, current, instance, host.Personality)
+	oldDelta := instance.Status.Delta
+
+	if inSync {
```

Comment:
The logic here looks good, can we wrap it into a new method? I'm wondering if we need to care about the other controllers. And I'll prefer a new unit test against the new method if possible

Reply:
made it generic. I'll raise a new issue to replace the other controllers status.delta with the SetInstanceDelta function